module/motion/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -0,0 +1,286 @@
+                bool walkableState = true;  // TODO: what is our equivalent here?
+
+                // see if the walk engine has new goals for us
+                walkableState = walk_engine.updateState(dt, current_orders, walkableState);
+
+                if ((walk_engine.getState() != engine::WalkEngineState::IDLE) && walkableState) {  // todo
+                    calculateJointGoals();
+                }
+            }
+        });
+    }  // namespace motion
+
+    float QuinticWalk::getTimeDelta() {
+        // compute time delta depended if we are currently in simulation or reality
+        float dt;
+        auto current_time = NUClear::clock::now();
+        // only take real time difference if walking was not stopped before
+        // TODO Is this really necessary?
+        auto time_diff_ms = std::chrono::duration_cast<std::chrono::milliseconds>(current_time - last_update_time);
+        dt                = time_diff_ms.count() / 1000.0f;
+        if (dt == 0.0f) {
+            // log<NUClear::WARN>(fmt::format("dt was 0 ({})", time_diff_ms.count()));
+            dt = 0.001f;
+        }
+        last_update_time = current_time;
+        // time is wrong when we run it for the first time
+        if (first_run) {
+            first_run = false;
+            dt        = 0.0001f;
+        }
+        return dt;
+    }
+
+    void QuinticWalk::calculateJointGoals() {
+        /*
+        This method computes the next motor goals and publishes them.
+        */
+        auto setRPY = [&](const float& roll, const float& pitch, const float& yaw) {
+            const float halfYaw   = yaw * 0.5f;
+            const float halfPitch = pitch * 0.5f;
+            const float halfRoll  = roll * 0.5f;
+            const float cosYaw    = std::cos(halfYaw);
+            const float sinYaw    = std::sin(halfYaw);
+            const float cosPitch  = std::cos(halfPitch);
+            const float sinPitch  = std::sin(halfPitch);
+            const float cosRoll   = std::cos(halfRoll);
+            const float sinRoll   = std::sin(halfRoll);
+            return Eigen::Quaternionf(cosRoll * cosPitch * cosYaw + sinRoll * sinPitch * sinYaw,  // formerly yzx
+                                      sinRoll * cosPitch * cosYaw - cosRoll * sinPitch * sinYaw,  // x
+                                      cosRoll * sinPitch * cosYaw + sinRoll * cosPitch * sinYaw,  // y
+                                      cosRoll * cosPitch * sinYaw - sinRoll * sinPitch * cosYaw   // z
+                                      )
+                .normalized()
+                .toRotationMatrix();
+        };
+
+        // Read the cartesian positions and orientations for trunk and fly foot
+        walk_engine.computeCartesianPosition(trunk_pos, trunk_axis, foot_pos, foot_axis, is_left_support);
+
+        // Change goals from support foot based coordinate system to trunk based coordinate system
+        Eigen::Affine3f Hst;  // trunk_to_support_foot_goal
```

Comment:
```suggestion
        Eigen::Affine3f Hst = Eigen::Affine3f::Identity();  // trunk_to_support_foot_goal
```

Reply:
Is there a point to assigning it the identity if the rotation and linear are immediately set afterwards?

Reply:
I believe that setting the translation sets the right-hand column, and linear is the upper 3x3, but this might have something to do with the 0,0,0,1 at the bottom of the matrix? However, I think that the default constructor of Affine3f is already identity but you would have to confirm.

Reply:
based on the Eigen source, it looks like the default constructor sets the last row to `{0, 0, 0, 1}` automatically.
(https://gitlab.com/libeigen/eigen/-/blob/master/Eigen/src/Geometry/Transform.h line 258)

Reply:
I remember having issues in the kinematics code that required me to set it to the identity, but that may have been for a different reason

Reply:
I don't think it sets anything except the bottom row - I wonder if the other rows were causing the issues? do you remember which PR or files you had that problem?

Reply:
If it's only the 0,0,0,1 that's set then if you don't set the translation and linear it would have junk in it besides that final row. That could have been the problem that was occuring

Reply:
Yea, I was using a running transform in ForwardKinematics, that was the issue. We should be fine without making this change.